src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/MetricsPipeline.cs

```diff
@@ -39,14 +40,24 @@ internal class MetricsPipeline : EventSourcePipeline<MetricsPipelineSettings>
 
         protected override MonitoringSourceConfiguration CreateConfiguration()
         {
+            bool useSharedSession = false;
```

Comment:
For correctness, I would further pull this out and add a `UseSharedSession` property to `MetricsPipelineSettings`; let the caller ultimately decided whether shared sessions are supported.

Reply:
Resolving within the Dotnet Monitor PR